open-hackathon-adminUI/src/app/admin/user_mgr.py

```diff
@@ -0,0 +1,160 @@
+            if u is not None:
+                r.online = u.user.online
+            else:
+                r.online = 0
+            return r
+
+        map(lambda r: online(r), reg_list)
+        return reg_list
+
+    def db_logout(self, user):
+        try:
+            self.db.update_object(user, online=0)
+            self.db.commit()
+            return "OK"
+        except Exception as e:
+            log.error(e)
+            return "log out failed"
+
+    def db_login(self, openid, **kwargs):
+        # update db
+        email_info = kwargs['email_info']
+        admin = self.db.find_first_object(Admin, openid=openid)
+        if admin is not None:
+            self.db.update_object(admin,
+                                  name=kwargs["name"],
+                                  nickname=kwargs["nickname"],
+                                  access_token=kwargs["access_token"],
+                                  avatar_url=kwargs["avatar_url"],
+                                  last_login_time=datetime.utcnow(),
+                                  online=1)
+            for n in range(0, len(email_info)):
+                email = email_info[n]['email']
+                primary_email = email_info[n]['primary']
+                verified = email_info[n]['verified']
+                if self.db.find_first_object(AdminEmail, email=email) is None:
+                    adminemail = AdminEmail(name=kwargs['name'], email=email, primary_email=primary_email,
+                                          verified=verified, admin=admin)
+                    self.db.add_object(adminemail)
+            self.db.commit()
+        else:
+            admin = Admin(openid=openid,
+                        name=kwargs["name"],
+                        nickname=kwargs["nickname"],
+                        access_token=kwargs["access_token"],
+                        avatar_url=kwargs["avatar_url"],
+                        online=1)
+
+            self.db.add_object(admin)
+            self.db.commit()
+
+            for n in email_info:
+                email = n['email']
+                primary_email = n['primary']
+                verified = n['verified']
+                useremail = AdminEmail(name=kwargs['name'], email=email, primary_email=primary_email,
+                                      verified=verified, admin=admin)
+                self.db.add_object(useremail)
+                self.db.commit()
+
+        # generate API token
+        token = self.__generate_api_token(admin)
+        return {
+            "token": token,
+            "user": admin
+        }
+
+    def validate_request(self):
+        if HTTP_HEADER.TOKEN not in request.headers:
+            return False
+
+        user = self.__validate_token(request.headers[HTTP_HEADER.TOKEN])
+        if user is None:
+            return False
+
+        g.user = user
+        return True
+
+    def get_user_by_id(self, user_id):
```

Comment:
refresh the whole user_mgr.py 
change "user" to "admin" in codes , as well as the logics in functions